driver-core/src/main/com/mongodb/internal/binding/AsyncClusterBinding.java

```diff
@@ -96,29 +97,31 @@ public void getConnectionSource(final ServerAddress serverAddress, final SingleR
 
     private void getAsyncClusterBindingConnectionSource(final ServerSelector serverSelector,
                                                         final SingleResultCallback<AsyncConnectionSource> callback) {
-        cluster.selectServerAsync(serverSelector, new SingleResultCallback<Server>() {
+        cluster.selectServerAsync(serverSelector, new SingleResultCallback<ServerTuple>() {
             @Override
-            public void onResult(final Server result, final Throwable t) {
+            public void onResult(final ServerTuple result, final Throwable t) {
                 if (t != null) {
                     callback.onResult(null, t);
                 } else {
-                    callback.onResult(new AsyncClusterBindingConnectionSource(result), null);
+                    callback.onResult(new AsyncClusterBindingConnectionSource(result.getServer(), result.getServerDescription()), null);
```

Comment:
I only added the tuple class because Java doesn't let you return multiple values from a method.    I didn't see the need to use it anywhere else.  Do you think it adds clarity to pass it around?

Reply:
I think given that the server.getDescription method has been removed its safe enough.